rust/src/delta_datafusion.rs

```diff
@@ -510,6 +529,59 @@ impl ExecutionPlan for DeltaScan {
     }
 }
 
+fn get_null_of_arrow_type(t: &ArrowDataType) -> ScalarValue {
+    match t {
+        ArrowDataType::Null => ScalarValue::Null,
+        ArrowDataType::Boolean => ScalarValue::Boolean(None),
+        ArrowDataType::Int8 => ScalarValue::Int8(None),
+        ArrowDataType::Int16 => ScalarValue::Int16(None),
+        ArrowDataType::Int32 => ScalarValue::Int32(None),
+        ArrowDataType::Int64 => ScalarValue::Int64(None),
+        ArrowDataType::UInt8 => ScalarValue::UInt8(None),
+        ArrowDataType::UInt16 => ScalarValue::UInt16(None),
+        ArrowDataType::UInt32 => ScalarValue::UInt32(None),
+        ArrowDataType::UInt64 => ScalarValue::UInt64(None),
+        ArrowDataType::Float32 => ScalarValue::Float32(None),
+        ArrowDataType::Float64 => ScalarValue::Float64(None),
+        ArrowDataType::Date32 => ScalarValue::Date32(None),
+        ArrowDataType::Date64 => ScalarValue::Date64(None),
+        ArrowDataType::Binary => ScalarValue::Binary(None),
+        ArrowDataType::FixedSizeBinary(size) => ScalarValue::FixedSizeBinary(size.to_owned(), None),
+        ArrowDataType::LargeBinary => ScalarValue::LargeBinary(None),
+        ArrowDataType::Utf8 => ScalarValue::Utf8(None),
+        ArrowDataType::LargeUtf8 => ScalarValue::LargeUtf8(None),
+        ArrowDataType::Decimal128(precision, scale) => {
+            ScalarValue::Decimal128(None, precision.to_owned(), scale.to_owned())
+        }
+        ArrowDataType::Timestamp(unit, tz) => {
+            let tz = tz.to_owned();
+            match unit {
+                TimeUnit::Second => ScalarValue::TimestampSecond(None, tz),
+                TimeUnit::Millisecond => ScalarValue::TimestampMillisecond(None, tz),
+                TimeUnit::Microsecond => ScalarValue::TimestampMicrosecond(None, tz),
+                TimeUnit::Nanosecond => ScalarValue::TimestampNanosecond(None, tz),
+            }
+        }
+        //Unsupported types...
+        ArrowDataType::Float16
+        | ArrowDataType::Decimal256(_, _)
+        | ArrowDataType::Union(_, _, _)
+        | ArrowDataType::Dictionary(_, _)
+        | ArrowDataType::LargeList(_)
+        | ArrowDataType::Struct(_)
+        | ArrowDataType::List(_)
+        | ArrowDataType::FixedSizeList(_, _)
+        | ArrowDataType::Time32(_)
+        | ArrowDataType::Time64(_)
+        | ArrowDataType::Duration(_)
+        | ArrowDataType::Interval(_)
+        | ArrowDataType::RunEndEncoded(_, _)
+        | ArrowDataType::Map(_, _) => {
+            panic!("{}", format!("Implement data type for Delta Lake {}", t));
```

Comment:
We usually try to be a panic-free crate, would it be possible to surface this error?

Reply:
It's unfortunately mostly used at an interface boundary with Datafusion that doesn't return a result. I removed the panic and restored the original behavior of returning a null scalar when it cannot be handled.